UndertaleModLib/Models/UndertaleTextureGroupInfo.cs

```diff
@@ -0,0 +1,146 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace UndertaleModLib.Models
+{
+    public class UndertaleTextureGroupInfo : UndertaleObject
+    {
+        public UndertaleString GroupName;
+        public List<UndertaleResourceById<UndertaleEmbeddedTexture>> TexturePages;
+        public List<UndertaleResourceById<UndertaleSprite>> Sprites;
+        public List<UndertaleResourceById<UndertaleSprite>> SpineSprites;
+        public List<UndertaleResourceById<UndertaleFont>> Fonts;
+        public List<UndertaleResourceById<UndertaleBackground>> Tilesets;
+
+        public UndertaleTextureGroupInfo()
+        {
+            TexturePages = new List<UndertaleResourceById<UndertaleEmbeddedTexture>>();
+            Sprites = new List<UndertaleResourceById<UndertaleSprite>>();
+            SpineSprites = new List<UndertaleResourceById<UndertaleSprite>>();
+            Fonts = new List<UndertaleResourceById<UndertaleFont>>();
+            Tilesets = new List<UndertaleResourceById<UndertaleBackground>>();
+        }
+
+        public void Serialize(UndertaleWriter writer)
+        {
+            writer.WriteUndertaleString(GroupName);
+            uint pointer1 = writer.Position; writer.Write(0);
```

Comment:
Stuff like this was kinda supposed to be abstracted away into UndertaleReader/Writer to not clutter up code with manual pointer magic like this... (I also have to kinda fix it a bit to better handle non-standard pointers e.g. messed up by other tools, but that's another story). You can use WriteUndertaleObjectPointer as long as you make the target implement UndertaleObject (also see my note about lists which should handle that)